### PR TITLE
aaru/ age zip script

### PR DESCRIPTION
**Pull request by AB3000**
References pyxeda/AiCLUB#159 (blocker bug, where datasets are outdated)

### CODE REVIEW

File: Guess_My_Age/data_generation_age.py [hunk 1]
```diff
@@ -81,24 +85,56 @@ def data_generator(fraction=0.5, total_samples=2001, out_file = "medium_data.csv
...
+     # data_generator(0.5, 2500, out_file="age_project_3_50_percent_adults.csv")
+     # data_generator(0.75, 2500, out_file="age_project_3_75_percent_adults.csv")
 
-    # #project 3 DATASETS
-    data_generator(0.0004, 2500, out_file="age_3_project_one_adult.csv") #only one adult example
-    data_generator(0.1, 2500, out_file="age_project_3_10_percent_adults.csv")
-    data_generator(0.3, 2500, out_file="age_project_3_30_percent_adults.csv")
-    data_generator(0.5, 2500, out_file="age_project_3_50_percent_adults.csv")
-    data_generator(0.75, 2500, out_file="age_project_3_75_percent_adults.csv")
+     # #project 4 DATASET
```
Comment (swamisundar) [inline Guess_My_Age/data_generation_age.py]: if you arent using it -- please remove it

File: utils/file_compression.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import glob
+import zlib
+import zipfile
+import os
+
+def compress_files(regex="age_project*", out_file="age_project.zip"):
```
Comment (swamisundar) [inline utils/file_compression.py]: please don't have default arguements

File: utils/file_compression.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+     compression = zipfile.ZIP_DEFLATED
+     # create the zip file first parameter path/name, second mode
+     zf = zipfile.ZipFile(out_file, mode="w") 
+
+     try:
+          for file_name in files:
+               # Add file to the zip file
+               # first parameter file to zip, second filename in zip
+               zf.write(file_name, file_name, compress_type=compression)
+               os.remove(file_name) #remove file from data/ directory because already added to zipfile
```
Comment (swamisundar) [inline utils/file_compression.py]: lets not delete the source files.

File: Guess_My_Age/data_generation_age.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import numpy as np
 import pandas as pd
+import random
```
Comment (swamisundar) [inline Guess_My_Age/data_generation_age.py]: from /utils import file_compression

Comment (AB3000) [reply Guess_My_Age/data_generation_age.py]: I created a top-level script instead that handles all the compression. Is that ok?

File: utils/file_compression.py (hunk 2 above)
Comment (AB3000) [reply utils/file_compression.py]: Does this apply for everything or just the compress_files function? For now I left the others and only deleted this one for the re-review.

File: utils/file_compression.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+     finally:
+          zf.close() # Don't forget to close the file!
+
+     print(out_file + " has been zipped") # let user know file has been zipped
+
+
+if __name__ == "__main__":
+     
+     if len(sys.argv) != 3:
+          print("Please enter a path to the files, regex, and output zipfile.\nUsage: python3 file_compression.py <path_to_files> <regex> <out_zipfile>")
```
Comment (swamisundar) [inline utils/file_compression.py]: break it into 2 lines.

**Pull request merged**